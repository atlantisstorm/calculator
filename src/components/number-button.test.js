import React from 'react';
import { render, fireEvent } from "@testing-library/react";
import NumberButton from './number-button';

describe('NumberButton component', () => {
  it('should match snapshot', () => {
    const onClick = jest.fn();
    const value = "5";

    const { asFragment } = render(
      <NumberButton onClick={ onClick } value={ value } />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should call dispatch with expected parameters', () => {
    const onClick = jest.fn();
    const value = "5";

    const { container, getByTestId } = render(
      <NumberButton onClick={ onClick } value={ value } />
    );

    const coreButton = getByTestId('core-button-5');
    fireEvent.click(coreButton);
    expect(onClick).toHaveBeenCalledTimes(1);
    expect(container.innerHTML).toContain("class=\"number\"");

    // Plus loads of other tests...
  });
});