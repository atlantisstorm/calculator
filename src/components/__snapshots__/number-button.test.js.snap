// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumberButton component should match snapshot 1`] = `
<DocumentFragment>
  <button
    class="number"
    data-testid="core-button-5"
    value="5"
  >
    5
  </button>
</DocumentFragment>
`;
